react/src/pages/utils.tsx

```diff
@@ -173,24 +173,56 @@ export function jsonToAnalyses(data: Array<any>): Analysis[] {
                 row.state = null;
                 break;
         }
-        return { ...row, selected: false } as Analysis;
+        return { ...row } as Analysis;
     });
     return rows;
 }
 
+/**
+ * Get the index of a material-table row.
+ * If a material-table row is not provided, return null.
+ */
+export function getRowIndex(row: any): number | null {
+    try {
+        let index: number = row.tableData.id;
+        return index;
+    } catch (error) {
+        console.error(error);
+        return null;
+    }
+}
+
+/**
+ * Return whether this material-table row is checked / selected.
+ * If it is not a material-table row, return null.
+ */
+export function getIsRowSelected(row: any): boolean | null {
+    try {
+        let isChecked: boolean = row.tableData.checked;
```

Comment:
You can use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) here (`?.` operator) and avoid the try-catch. If it's just a boolean then you can really just `return `row?.tableData?.checked`. This and `getRowIndex` might not need to be separate functions out here then if it boils down to that. I'm guessing `material-table` doesn't export `tableData` in its typings?

Reply:
Yeah tableData is not exported. Also strangely if the row is unchecked, then the `checked` field is just undefined instead of false.

Reply:
You can use `!!row?.tableData?.checked` to coerce to boolean then